### DISCUSSION TITLE
Few questions if I may

### DISCUSSION THREAD

**Discussion by galosre** [tesla-android/issue-tracker#21] [General]
Got "Virtual touchscreen is not active, wait for Android to boot up" the problem I think Android is active but no touchscreen!
After this, I got Page unresponsive and Tesla browser asked to exit
Where is Autokit do I need the carlink dongle to enable it?
In the meantime, can we connect the phone to use it as gps?
I was curious I downloaded Waze directly from play store it opened, but the whole display became vertical on the side mean all the icon instead to be at the bottom where on the right vertical side

**Comment by mikegapinski** [comment]
If touchscreen is not active your Android Pi does not have a proper IP address. Please check if internet connectivity is present on Tesla Android wifi network.

**Comment by galosre** [comment]
Before got internet connectivity With LTE pluged into RPI Linux I double-check the Ethernet cable it was working, I even used another one to make sure.
So I decided to reconfigure the pi-hole static address but now after that impossible to obtain  an IP at Wi-Fi connection
![2022-05-17_201126](https://user-images.githubusercontent.com/22267106/168949831-73d062ba-ac99-4251-a29b-908e6cbed0ba.jpg)
!

**Reply by galosre** [reply]
I noticed my rpi3 did not have any green light on the Ethernet port although some says is  meant connection at 100mbs anyway I had another one  I replaced it , I did the SD again, got  much faster to get IP  with Wi-Fi + internet but unfortunately still no touchscreen. I will get the dongle from Carlinkit tomorrow, but I do not think it has to do anything with the touchscreen

**Reply by blu31c33** [reply]
im having the exact same issue. same error message. no touchscreen. it worked fine until my carlinkit adapter did a firmware update. carplay shows up fine so its odd it would be touchscreen impacted by a carlinkit update.

**Reply by blu31c33** [reply]
i resolved this issue by re-configuring pi-hole.  for some reason i lost the DHCP reservation/mac address i put in (3.3.3.50).  once i updated that (and it stuck on a reboot), the issue with the error message and touchscreen no longer working was resolved.  hope that helps.

**Comment by galosre** [comment]
Thank you for showing me the way, to start  I did not click the + green button (not in the instruction I think) but I had to do it at least 5 times since pi-hole kept changing the mac address or Ip, and finally it took it and **yes** I got touch screen also major power issue from the tesla to the PI3 no (red led) something I need to address maybe an other cable ( recommendation?)

**Comment by galosre** [comment]
Everything working contribution has been  sent
Thank you
Philip

**Comment by galosre** [comment]
Well after checking got more issues
- First, no sound at all. It is not clear about Bluetooth connection got no navigation sound and or Amazon/ YouTube Music etc... I tried Bluetooth phone to android 12 pair but  cannot connect
- During driving, got several times white page I had to refresh or black page with no connection in big letter!
- For me I resolved the power issue  by not using the USB C plug but instead the cigarette lighter I plugged with a warp dash adapter (oneplus) since I got this phone but probably any PD adapter with dualport  would work.
